sdk/eventhubs/azure-messaging-eventhubs/inc/azure/messaging/eventhubs/eventhubs_exception.hpp

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include <azure/core/amqp/models/amqp_error.hpp>
+
+#include <stdexcept>
+#include <string>
+
+namespace Azure { namespace Messaging { namespace EventHubs {
+
+  enum class EventHubsStatusCode : int32_t
+  {
+    Invalid,
+    Ok,
+    Error,
+    Timeout,
+    Cancelled,
+  };
+
+  /**
+   * @brief An exception thrown when an EventHubs service operation fails.
+   */
+  struct EventHubsException final : std::runtime_error
+  {
+    /**
+     * @brief Constructs a #EventHubsException with a message.
+     *
+     * @param what An explanatory string.
+     */
+    explicit EventHubsException(const std::string& what)
+        : std::runtime_error(what), ErrorDescription{what}
+    {
+    }
+
+    /**
+     * @brief Constructs a #EventHubsException with an error condition.
+     *
+     * @param error The AMQP Error indicating the error.
+     */
+    explicit EventHubsException(Azure::Core::Amqp::Models::_internal::AmqpError const& error)
+        : std::runtime_error(error.Description), ErrorCondition(error.Condition.ToString()),
```

Comment:
Another public type/method with internal concept exposed.

Reply:
That was *painful*.